core/src/main/scala/kafka/log/OffsetIndex.scala

```diff
@@ -185,6 +185,7 @@ class OffsetIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writabl
       _entries = entries
       mmap.position(_entries * entrySize)
       _lastOffset = lastEntry.offset
+      debug(s"Truncated index ${file.getAbsolutePath} to $entries entries; position is now ${mmap.position()} and last offset is now ${_lastOffset}")
```

Comment:
nit: we're a little inconsistent on use of `file.getName` vs `file.getAbsolutePath`. Also, it would be nice to consistently favor the use of `${}` substitutions over `format`.

Reply:
Went with `file.getAbsolutePath` in order to identify the partition, since we don't have a LogContext. Also replaced